go/billing/api.py

```diff
@@ -493,3 +510,32 @@ def getChild(self, name, request):
     def render_GET(self, request):
         request.setResponseCode(200)  # OK
         return ''
+
+    @defer.inlineCallbacks
+    def health_check(self, request):
+        """
+        We want our health check to be comprehensive, so we implement it here.
+        """
+        try:
+            yield self._connection_pool.runQuery("SELECT 1")
+        except (ConnectionDead, psycopg2.InterfaceError):
+            request.setResponseCode(503)
+            request.write("Database connection unavailable\n")
+        else:
+            request.setResponseCode(200)
+            request.write("OK\n")
+        request.finish()
```

Comment:
I put the request writing there to avoid another layer of indirection, but it doesn't buy much and it does give the request object to things that shouldn't really have it. Thanks.